src/core/CMakeLists.txt

```diff
@@ -60,6 +58,10 @@ else ()
             )
 endif ()
 
+if(NGEN_WITH_MPI)
+    target_link_libraries(core PUBLIC MPI::MPI_C MPI::MPI_CXX)
```

Comment:
Does this need a nested check for Fortran as well, to pull in the MPI Fortran bindings?

Reply:
This linking is for `HY_Features_MPI`, so I don't think so